controller/nodegroup.go

```diff
@@ -253,20 +253,35 @@ func createNodeGroup(config *eksv1.EKSClusterConfig, group eksv1.NodeGroup, eksS
 		nodeGroupCreateInput.Subnets = aws.StringSlice(config.Status.Subnets)
 	}
 
-	finalTemplate := fmt.Sprintf(templates.NodeInstanceRoleTemplate, getEC2ServiceEndpoint(config.Spec.Region))
-	output, err := createStack(svc, fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName), config.Spec.DisplayName, finalTemplate, []string{cloudformation.CapabilityCapabilityIam}, []*cloudformation.Parameter{})
-	if err != nil {
-		return "", err
+	if aws.StringValue(group.NodeRole) == "" {
+		if config.Status.GeneratedNodeRole == "" {
+			finalTemplate := fmt.Sprintf(templates.NodeInstanceRoleTemplate, getEC2ServiceEndpoint(config.Spec.Region))
+			output, err := createStack(svc, fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName), config.Spec.DisplayName, finalTemplate, []string{cloudformation.CapabilityCapabilityIam}, []*cloudformation.Parameter{})
+			if err != nil {
+				// If there was an error creating the node role stack, return an empty launch template
+				// version and the error.
+				return "", err
+			}
+
+			// Ensure the change persists for the calling function. No need to save here because the
+			// config object will get saved by the calling function.
+			*config = *config.DeepCopy()
+			config.Status.GeneratedNodeRole = getParameterValueFromOutput("NodeInstanceRole", output.Stacks[0].Outputs)
```

Comment:
Where does this get saved? I think it should probably be added here: https://github.com/rancher/eks-operator/pull/73/files#diff-a74e30a65934de22a0dafc6fd3a31ff5c5c81d47e8fbd43f4f740c419977b565R1155, and this function should probably not write to the config at all as it is a side effect of what it's main goal is, creating the nodegroup in eks.